### PR TITLE
Multicurrency support

### LINKED ISSUE #8: Multicurrency support
Each `Spree::Order` [has a currency associated](https://github.com/spree/spree/blob/master/core/app/models/spree/order.rb#L181), which we currently don't use when creating a payment.

Since Mollie now supports Multicurrency, we should use the currency associated with the order when creating a new Mollie Payment.

Once `mollie-api-ruby` is updated to support the V2 API, MC can be implemented for Spree as well.

To do:

- [x] Update `mollie-api-ruby` dependency to support MC.
- [ ] Release as breaking change (version 2.0.0).

cc @tomasmollie

### PR DESCRIPTION
**Pull request by vernondegoede**
Fixes #8.

- [x] Set current currency and amount.
- [x] Get available payment methods based on current currency + amount
- [x] Update payment states (`cancelled` => `canceled`).
- [x] Pass amount object (including currency) when creating a new refund.
- [x] Update method constants.
- [x] Update Spree API endpoint (methods)
- [x] Update tests.

### CODE REVIEW

**Comment by vernondegoede** [comment]
@justincase Could you take a look at the code?

File: app/models/spree/gateway/mollie_gateway.rb [hunk 1]
```diff
@@ -72,9 +72,13 @@ def prepare_payment_params(money_in_cents, source, gateway_options)
       order_number = gateway_options[:order_id]
       customer_id = gateway_options[:customer_id]
       amount = money_in_cents / 100.0
+      currency = gateway_options[:currency]
 
       order_params = {
-          amount: amount,
+          amount: {
+              value: sprintf("%.2f", amount.to_f),
```
Comment (justincase) [inline app/models/spree/gateway/mollie_gateway.rb]: This might be an issue for `JPY`?

File: app/models/spree/gateway/mollie_gateway.rb [hunk 2]
```diff
@@ -164,7 +190,7 @@ def update_by_mollie_status!(mollie_payment, payment)
           payment.complete! unless payment.completed?
           payment.order.finalize!
           payment.order.update_attributes(:state => 'complete', :completed_at => Time.now)
-        when 'cancelled', 'expired', 'failed'
+        when 'canceled', 'expired', 'failed'
           payment.failure! unless payment.failed?
         when 'refunded'
```
Comment (justincase) [inline app/models/spree/gateway/mollie_gateway.rb]: `refunded` status has been removed in `v2`.

File: app/models/spree/gateway/mollie_gateway.rb (hunk 1 above)
Comment (vernondegoede) [reply app/models/spree/gateway/mollie_gateway.rb]: Thanks, yes it is. I'm now using the money object to always use the correct formatting.

File: app/models/spree/gateway/mollie_gateway.rb (hunk 2 above)
Comment (vernondegoede) [reply app/models/spree/gateway/mollie_gateway.rb]: Removed it ✅

**Pull request merged**